src/account_manager/operations/address_generation.rs

```diff
@@ -0,0 +1,144 @@
+                            account_index,
+                            WalletEvent::LedgerAddressGeneration(AddressData {
+                                address: address[0].to_bech32(bech32_hrp.clone()),
+                            }),
+                        );
+                    }
+
+                    // Generate with prompt so the user can verify
+                    ledger_nano
+                        .generate_addresses(
+                            self.coin_type.load(Ordering::Relaxed),
+                            account_index,
+                            address_index..address_index + 1,
+                            internal,
+                            options,
+                        )
+                        .await?
+                } else {
+                    ledger_nano
+                        .generate_addresses(
+                            self.coin_type.load(Ordering::Relaxed),
+                            account_index,
+                            address_index..address_index + 1,
+                            internal,
+                            options,
+                        )
+                        .await?
+                }
+            }
+            #[cfg(feature = "stronghold")]
+            SecretManager::Stronghold(stronghold) => {
+                stronghold
+                    .generate_addresses(
+                        self.coin_type.load(Ordering::Relaxed),
+                        account_index,
+                        address_index..address_index + 1,
+                        internal,
+                        options,
+                    )
+                    .await?
+            }
+            SecretManager::Mnemonic(mnemonic) => {
+                mnemonic
+                    .generate_addresses(
+                        self.coin_type.load(Ordering::Relaxed),
+                        account_index,
+                        address_index..address_index + 1,
+                        internal,
+                        options,
+                    )
+                    .await?
+            }
+            SecretManager::Placeholder(_) => return Err(iota_client::Error::PlaceholderSecretManager.into()),
+        };
+
+        Ok(*address.first().ok_or(crate::Error::MissingParameter("address"))?)
+    }
+
+    // Get the bech32 hrp from the first account address or if not existent, from the client
+    #[allow(dead_code)]
+    pub(crate) async fn get_bech32_hrp(&self) -> crate::Result<String> {
+        Ok(match self.get_accounts().await?.first() {
+            Some(account) => account
+                .public_addresses()
+                .await
+                .first()
+                .expect("missing first public address")
+                .address
+                .bech32_hrp
+                .clone(),
+            None => {
+                self.client_options
+                    .read()
+                    .await
+                    .clone()
+                    .finish()?
```

Comment:
Are we creating a new client here ?

Reply:
Yes if we don't have an account, then we also don't have a client and need to create it new (unless we do some bigger change and create the client in the AccountManager already)

Reply:
Created an issue, I don't like that we are randomly creating clients here and there